(blog) 2021-01-29 status update (#185)
puppetlabs/iac: _posts/2021-01-29-status-update.md

author: tphoney
categories:
  - team
  - status
tags:
  - litmus
  - cloud-ci
  - modules
  - iac-team
  - community-day
---

Hey everyone, it's my honour to write my first blogpost for Puppet. Kinda strange in [I][TP] have done a number of other things for puppet, but never a blogpost. Case in point, my talk about [litmus](https://www.youtube.com/watch?v=FYfR7ZEGHoE) (available permanently on youtube to make me shudder at my own voice). On to business !!!

## What is the IAC Team

How did you find this post then ? Just kidding, we are a team at puppet that looks after all the modules / tasks / gems / tools that you as a community use. If we dont look after it we know who you should talk to. Here is a great blog post on how to reach us [Puppet IAC Team](https://puppetlabs.github.io/iac/team/2021/01/20/reaching-out.html)

## Ecosystem Kickoff

The IAC team is part of a larger group at puppet that has as its main focus 'you' the community. We had our opening meeting discussing the direction of our work for the next year, and it all looks very exciting. I can't give away too much yet, but more details will trickle out as we confirm our goals.

## Automated Module Release

> What I can talk a little about is [Danny's][Danny] work on a new proof of concept. We are an automation company so we are always looking to remove a few steps from our process, this project will enable us to automate some of the manual steps required to release modules. Leaving the the human to only review and click merge on Github.

Reviewer: @carabasdaniel , do you have a link to the work handy, so that folks can have a peek on what's cooking?

## DSC

> Everyone's favourite topic, [Paula][Paula] and [Michael][Michael] have hit their first milestone, and are now focusing on the most important part. Documentation and blogposts ! There will also be a rebuild of the DSC modules to fix a vendored path issue. AND the march to 1.0 is well underway with a real focus on testing.

Reviewer: link `milestone` to https://github.com/puppetlabs/Puppet.Dsc/milestone/3?closed=1 ?

## Networking Device Support Enhancements

[Ciaran][Ciaran] has been working on enhancing the [cisco_ios](https://github.com/puppetlabs/cisco_ios) module to include two new [Bolt tasks](https://puppet.com/docs/bolt/latest/writing_tasks.html) to backup and restore configurations. The PoC has been proven out and now we're just wrapping up on some final development tasks before moving on to more extensive testing. We are looking to make this as universal as we can across as many Cisco devices as we can and have some exciting developments in the pipeline for better testing against Cisco devices. More details soon...

## Community Contributions

On the community day we normally focus on all the wonderful contributions we receive in Github, but over the past couple of weeks we have started to look at our JIRA backlog. There are a lot of tickets there as you can imagine. We have been categorising them based on the module they are associated with and also closing off tickets that are no longer relevant or that have already been fixed.

> 

Reviewer 2: May be we can give the actual numbers of PR and tickets looked on community day.

IAC Team processed 19 community PRs processed (9 tooling, 11modules) also we have looked at 31 module issues.

Author: that information is from the slide sheet ?

Reviewer 2: yeah thats right its from the google sheet.

---
layout: post
title: "2021-01-29: IAC Team Status Update"
author: tphoney
categories:
  - team
  - status

> tags:

Reviewer 3: Just some minor alignment issues with the tags

- litmus
  - cloud-ci
  - modules
  - iac-team
  - community-day
---

> Hey everyone, its my honour to write my first blogpost for Puppet. Kinda strange in i have done a number of other things for puppet, but never a blogpost. Case in point, my talk about [litmus](https://www.youtube.com/watch?v=FYfR7ZEGHoE) (available permanently on youtube to make me shudder at my own voice). On to buisness !!!

Reviewer 3: ```suggestion
Hey everyone, it's my honour to write my first blogpost for Puppet. Kinda strange in [I][TP] have done a number of other things for puppet, but never a blogpost. Case in point, my talk about [litmus](https://www.youtube.com/watch?v=FYfR7ZEGHoE) (available permanently on youtube to make me shudder at my own voice). On to business !!!
```

## What is the IAC Team

How did you find this post then ? Just kidding, we are a team at puppet that looks after all the modules / tasks / gems / tools that you as a community use. If we dont look after it we know who you should talk to. Here is a great blog post on how to reach us [Puppet IAC Team](https://puppetlabs.github.io/iac/team/2021/01/20/reaching-out.html)

## Ecosystem Kickoff

The IAC team is part of a larger group at puppet that has as its main focus 'you' the community. We had our opening meeting discussing the direction of our work for the next year, and it all looks very exciting. I can't give away too much yet, but more details will trickle out as we confirm our goals.

## Automated Module Release

> What i can talk a little about is Danny's work on a new proof of concept. We are an automation company so we are always looking to remove a few steps from our process, this project will enable us to automate some of the manual steps required to release modules. Leaving the the human to only review and click merge on github.

Reviewer 3: ```suggestion
What I can talk a little about is [Danny's][Danny] work on a new proof of concept. We are an automation company so we are always looking to remove a few steps from our process, this project will enable us to automate some of the manual steps required to release modules. Leaving the the human to only review and click merge on Github.
```

## DSC

> Everyone's favourite topic, paula and michael have hit their first milestone, and are now focusing on the most important part. Documentation and blogposts ! There will also be a rebuild of the DSC modules to fix a vendored path issue. AND the march to 1.0 is well underway with a real focus on testing.

Reviewer 3: ```suggestion
Everyone's favourite topic, [Paula][Paula] and [Michael][Michael] have hit their first milestone, and are now focusing on the most important part. Documentation and blogposts ! There will also be a rebuild of the DSC modules to fix a vendored path issue. AND the march to 1.0 is well underway with a real focus on testing.
```

> 

Reviewer 3: I should have given you my update for this week too, so I'll do it here:

```suggestion

## Networking Device Support Enhancements

[Ciaran][Ciaran] has been working on enhancing the [cisco_ios](https://github.com/puppetlabs/cisco_ios) module to include two new [Bolt tasks](https://puppet.com/docs/bolt/latest/writing_tasks.html) to backup and restore configurations. The PoC has been proven out and now we're just wrapping up on some final development tasks before moving on to more extensive testing. We are looking to make this as universal as we can across as many Cisco devices as we can and have some exciting developments in the pipeline for better testing against Cisco devices. More details soon...
```

## Community Contributions

> On the community day we normally focus on all the wonderful contributions we recieve in github, but over the past couple of weeks we have started to look at our Jira backlog. There are a lot of tickets there as you can imagine. We have been catagorising them based on the module they are associated with and also closing off tickets that are no longer relavent or that have already been fixed.

Reviewer 3: ```suggestion
On the community day we normally focus on all the wonderful contributions we receive in Github, but over the past couple of weeks we have started to look at our JIRA backlog. There are a lot of tickets there as you can imagine. We have been categorising them based on the module they are associated with and also closing off tickets that are no longer relevant or that have already been fixed.
```

- [`puppetlabs-stdlib#1154`][puppetlabs-stdlib-pr-1154]: "Drop facter.d facts needed for Facter < 1.7", thanks to [ekohl][ekohl]
- [`puppetlabs-stdlib#1150`][puppetlabs-stdlib-pr-1150]: "stdlib::ensure: new fuction to cast ensure values", thanks to [b4ldr][b4ldr]
- [`pdk#938`][pdk-pr-938]: "(MAINT) Update Travis/Appveyor to build `main` branch", thanks to [scotje][scotje]
- [`puppet-modulebuilder#32`][puppet-modulebuilder-pr-32]: "Use Puppet 7 in development on Ruby 2.7+", thanks to [ekohl][ekohl]
- [`puppet-modulebuilder#31`][puppet-modulebuilder-pr-31]: "Add a setter for release_name", thanks to [ekohl][ekohl]
- [`puppetlabs_spec_helper#330`][puppetlabs_spec_helper-pr-330]: "Restructure PuppetLint rake tasks so they can be configurable", thanks to [nmaludy][nmaludy]

Please swing by our [office hours on Monday](https://puppet.com/community/office-hours/) on Slack if you want to directly contact us about anything module related.

## New Module / Gem Releases

The following modules were released this week:

- [`puppetlabs-apache`][puppetlabs-apache] (`5.9.0`)
- [`puppetlabs-postgresql`][puppetlabs-postgresql] (`6.9.0`)
- [`puppetlabs-iis`][puppetlabs-iis] (`7.2.0`)

> ## Our Open Source Contributors

Reviewer 3: I like this idea, we should integrate this in to the automated generation, actually 👍 

**EDIT:** Actually, reading this again, this may have been an inadvertent suggestion. I thought this was you calling out all the names on top of the Community Contributions section?

- [`puppetlabs-stdlib#1154`][puppetlabs-stdlib-pr-1154]: "Drop facter.d facts needed for Facter < 1.7", thanks to [ekohl][ekohl]
- [`puppetlabs-stdlib#1150`][puppetlabs-stdlib-pr-1150]: "stdlib::ensure: new fuction to cast ensure values", thanks to [b4ldr][b4ldr]
- [`pdk#938`][pdk-pr-938]: "(MAINT) Update Travis/Appveyor to build `main` branch", thanks to [scotje][scotje]
- [`puppet-modulebuilder#32`][puppet-modulebuilder-pr-32]: "Use Puppet 7 in development on Ruby 2.7+", thanks to [ekohl][ekohl]
- [`puppet-modulebuilder#31`][puppet-modulebuilder-pr-31]: "Add a setter for release_name", thanks to [ekohl][ekohl]
- [`puppetlabs_spec_helper#330`][puppetlabs_spec_helper-pr-330]: "Restructure PuppetLint rake tasks so they can be configurable", thanks to [nmaludy][nmaludy]

Please swing by our [office hours on Monday](https://puppet.com/community/office-hours/) on Slack if you want to directly contact us about anything module related.

## New Module / Gem Releases

The following modules were released this week:

- [`puppetlabs-apache`][puppetlabs-apache] (`5.9.0`)
- [`puppetlabs-postgresql`][puppetlabs-postgresql] (`6.9.0`)
- [`puppetlabs-iis`][puppetlabs-iis] (`7.2.0`)

> ## Our Open Source Contributors

Reviewer 3: The contributors on [lines 66-72](https://github.com/puppetlabs/iac/pull/185/files#diff-e8352d965ccd9240ebc2b60fb1cf24eb3febd14d1a035d54be8d7246f3368470R66-R72) won't actually show up here. Sorry, I should have spotted this earlier. 

Something like this perhaps:
```MD
## Our Open Source Contributors

  - [ekohl][ekohl]
  - [e-gris][e-gris]
  - [chillinger][chillinger]
  - [olevole][olevole]
  - [b4ldr][b4ldr]
  - [scotje][scotje]
  - [nmaludy][nmaludy]
```
